feat(docs-infra): change navigation in resources page

ajitsinghkaler opened the pull request:
https://angular.io/resources needs to be structured to be able to navigate to all resources with improved user experience. A lone scroll bar in this page will not help the reader a great deal in exploring the resources
Fixes #33526

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Earlier there was no navigation available on the resources page 

Issue Number: 33526


## What is the new behavior?
Added navigation to the resources page

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

ajitsinghkaler commented:
@gkalpak can you please look into this.

mary-poppins commented:
You can preview c771067 at https://pr34756-c771067.ngbuilds.io/.

gkalpak commented on aio/src/app/custom-elements/resource/resource-list.component.ts:
```diff
@@ -28,11 +29,19 @@ export class ResourceListComponent implements OnInit {
+  selectCategory(id: string) {
+    this.selectedCategory =
+      this.categories.find(category => category.id.toLowerCase() === id) || this.categories[0];
+  }
```
Can you please make this component similar to the [ContributorListComponent][1] (i.e. read/write from/to the URL query string, ignore case of `id`, etc.).

[1]: https://github.com/angular/angular/blob/c7710675e/aio/src/app/custom-elements/contributor/contributor-list.component.ts

gkalpak reviewed (changes requested):
Thx, @ajitsinghkaler! This looks great. A couple more tweaks and should be ready to merge (as far as I am concerned).

I think it now also makes sense to center the content under the tabs (because it looks weird to have the tabs centered and the content below left-aligned:

![resources-uncentered](https://user-images.githubusercontent.com/8604205/72339264-a2c1e380-36ce-11ea-884e-f08faadadd4e.png)

##
Note to self:
This should be approved by @fw-docs-marketing too (once ready).

ajitsinghkaler commented:
@gkalpak should I make the content below center-aligned or should I make it take 100% of the width

gkalpak commented:
I would preserve the current width.

mary-poppins commented:
You can preview 4a4f67f at https://pr34756-4a4f67f.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak can yo please review this.

gkalpak commented on aio/src/app/custom-elements/resource/resource-list.component.html:
```diff
@@ -1,28 +1,28 @@
-          <a class="h-anchor-offset" id="{{category.id}}"></a>
-          <h2>{{category.title}}</h2>
-        </header>
+  <div class="flex-center group-buttons">
```
Thx for fixing the indentation too ❤️

gkalpak commented on aio/src/app/custom-elements/resource/resource-list.component.ts:
```diff
@@ -14,10 +15,12 @@ export class ResourceListComponent implements OnInit {
-    private resourceService: ResourceService) {
+    private resourceService: ResourceService,
+    private locationService: LocationService) {
     this.location = location.pathname.replace(/^\/+/, '');
```
Not directly related to this PR, but ?I? just noticed that the `location` property (and thus the `PlatformLocation` dependency) and the `href()` method are not used. If that is the case, could you please create another commit (I would put it before the current commit) cleaning up the component by removing these unused bits? 🙏

gkalpak replied:
Can you please remove the `location` property (which doesn't seem to be used)?

gkalpak commented on aio/src/app/custom-elements/resource/resource-list.component.ts:
```diff
@@ -26,13 +29,24 @@ export class ResourceListComponent implements OnInit {
 
   @HostListener('window:scroll', ['$event.target'])
   onScroll(target: any) {
     this.scrollPos = target ? target.scrollTop || target.body.scrollTop ||  0 : 0;
```
Not directly related to this PR, but the `scrollPos` property and the `onScroll()` method don't seem to be used. Could you include them in the cleanup commit mentioned in a previous comment?

gkalpak reviewed (changes requested):
Thx, @ajitsinghkaler :+1: Yes, please, add some tests for the new functionality (i.e. toggling between categories, showing entries from a specific category only and interacting with the URL query).

I have left some comments to do some clean-up (which is not directly related to this PR, but would be awesome if you could do (on a separate commit)).

Other than that, this lgtm (pending the addition of tests) 👍

ajitsinghkaler committed:
refactor(docs-infra): remove unused functions in resources component page

In resources component there were unused functions removed unused funtions

mary-poppins commented:
You can preview 91ea8d6 at https://pr34756-91ea8d6.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak I've added the cleanup please have a look

gkalpak reviewed (changes requested):
The tests are failing, because the injector that is set up for the tests must be updated to provide the necessary dependencies [here][1]. I.e. remove `PlatformLocation` and provide a mock `Location` service (you can use [this one][2]).

<sub>Don't forget to add tests for the new functionality (as mentioned above).</sub>

[1]: https://github.com/angular/angular/blob/91ea8d6f3bb4bb9a2ac77839024f34ca52e03b39/aio/src/app/custom-elements/resource/resource-list.component.spec.ts#L21
[2]: https://github.com/angular/angular/blob/91ea8d6f3bb4bb9a2ac77839024f34ca52e03b39/aio/src/testing/location.service.ts

mary-poppins commented:
You can preview 42c2efc at https://pr34756-42c2efc.ngbuilds.io/.

mary-poppins commented:
You can preview 111f54d at https://pr34756-111f54d.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak added the tests please have alook

gkalpak reviewed (changes requested):
Almost there :smiley:
I left some minor clean-up suggestions for the tests. Other than that, you just need to get CI green and this is good to go.

mary-poppins commented:
You can preview 1a05786 at https://pr34756-1a05786.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak I've made all the changes. @sjtrimble can you please review my pull request

sjtrimble reviewed (approved):
Nice!

gkalpak reviewed (changes requested):
I just noticed that the commit messages need some clean up (to follow our [giudelines][1]).
Can you please squash the last two commits into one (so that the PR has overall two commits) and make sure there is a blank line above `Fixes ...` in the second commit message?

[1]: https://github.com/angular/angular/blob/c1cf46c5c/CONTRIBUTING.md#commit

ajitsinghkaler committed:
feat(docs-infra): change navigation in resources page

https://angular.io/resources needs to be sturctured to be able to navigate to all resources with improved user experience. A lone scroll bar in this page will not help the reader a great deal in exploring the resources

Fixes #33526

mary-poppins commented:
You can preview f7b0233 at https://pr34756-f7b0233.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak made the changes please have a look

ajitsinghkaler commented:
@gkalpak can you please add the merge label
